### ISSUE TITLE
Dotted underline on demo gets cut off when backspacing

### ISSUE THREAD

**Issue by 1j01** [yabwe/medium-editor#1031]
If you make a newline in the middle of the dotted underline on the demo and then backspace, the underline gets removed from the second part of the text.

![medium-editor dotted underline](https://cloud.githubusercontent.com/assets/3630663/14130131/9c560c52-f5fe-11e5-958a-79b624df7581.gif)

**Comment by nmielnik** [comment]
The dotted underline is actually added via some custom css just for the demo which is tied to a `<span>` with a class name.  When the newline is added, it splits up that `<span>` and the css is no longer applied to the newly created element.  

Maybe the css could be updated to not show up once someone starts editing...

**Comment by 1j01** [comment]
It's lost when joining lines, not splitting. But you can do whatever you want.

**Comment by nmielnik** [comment]
Yes, you're right.  In Chrome, it looks like the css class is actually copied over to the newly created `<span>` within a new `<p>` whenever you insert the newline (so the underline shows up when it's split)

<img width="441" alt="screen shot 2016-04-02 at 2 07 01 pm" src="https://cloud.githubusercontent.com/assets/2444240/14228188/6592d142-f8dc-11e5-8de2-a81995ca998e.png">

But then when you backspace and the contents of the two `<p>` are merged, only one of the two `<span>`s still have the class, the other has some inline styles added as well.

<img width="514" alt="screen shot 2016-04-02 at 2 10 52 pm" src="https://cloud.githubusercontent.com/assets/2444240/14228206/c516cce0-f8dc-11e5-93f6-b7a47768762b.png">

The issue also happens in Safari, but it doesn't appear to happen in Firefox or IE (in those browsers, when you backspace, the dotted underline remains).

Also, while I was playing around with it in Safari, it seems like something with spellcheck was firing and changing the text depending on where I inserted the newline:

**Before:**
<img width="502" alt="screen shot 2016-04-02 at 2 20 49 pm" src="https://cloud.githubusercontent.com/assets/2444240/14228270/298de41e-f8de-11e5-8a45-52ecdf2a5a98.png">

**After:**
<img width="486" alt="screen shot 2016-04-02 at 2 19 47 pm" src="https://cloud.githubusercontent.com/assets/2444240/14228274/3418199a-f8de-11e5-9322-77672dd929f6.png">

You can see it converted 'highli' into 'highly' just by adding a newline (no spellcheck prompt or anything).

So yeah, looks like there are some issues that could be fixed here, especially around the css-based styling (the dotted underline) clashing a bit with how the browser is handling styling of the text and merging elements together.
